Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -393,8 +393,9 @@ extension BrowserViewController: WKNavigationDelegate {
   public func webView(_ webView: WKWebView, decidePolicyFor navigationResponse: WKNavigationResponse, decisionHandler: @escaping (WKNavigationResponsePolicy) -> Void) {
     let response = navigationResponse.response
     let responseURL = response.url
+    let tab = tabManager[webView] ?? (webView as? TabWebView)?.tab
```

Comment:
Maybe if it's done every time across this file we can create a helper to always grab the proper tab? instead of writing it like this every time. I'm fine if we keep it and address it later in the rewrite